taichi/transforms/constant_fold.cpp

```diff
@@ -90,10 +142,65 @@ class ConstantFold : public BasicStmtVisitor {
   }
 };
 
+class ConstantFoldJIT : public BasicStmtVisitor {
+ public:
+  using BasicStmtVisitor::visit;
+  std::vector<BinaryOpStmt *> bops;
+  std::vector<UnaryOpStmt *> uops;
+
+  ConstantFoldJIT() : BasicStmtVisitor() {
+  }
+
+  void visit(UnaryOpStmt *stmt) override {
+    uops.push_back(stmt);
+  }
+
+  void visit(BinaryOpStmt *stmt) override {
+    bops.push_back(stmt);
```

Comment:
Please notice that we can only fold statements with all operands being `ConstStmt`s :)

And therefore, ultimately, we need to run it iteratively like `ConstantFold::run(IRNode *node)`.